pkg/controller/servicebindingrequest/controller.go

```diff
@@ -36,70 +34,58 @@ func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
 	return &Reconciler{client: mgr.GetClient(), dynClient: dynClient, scheme: mgr.GetScheme()}, nil
 }
 
-func addServiceBindingRequestWatches(c controller.Controller) error {
+// add adds a new Controller to mgr with r as the reconcile.Reconciler.
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	opts := controller.Options{Reconciler: r}
+	c, err := controller.New("servicebindingrequest-controller", mgr, opts)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Well, `getWatchingGVKs` is already returning a list, and that's all we need in that context. If we need to influence this list, and I believe we do in the near feature, then we can make `getWatchingGVKs` return a different list instead of breaking this logic in the middle of `add` method. Add method would only be reponsible by "adding" it does not need to differentiate which GVK.

Reply:
Please consider latest implementation, considering the discussions we had about the workflow design. We follow a stream of events, changes in Kubernetes, and transform those to become SBRs on the fly, based in annotations or the actual object. So there's a lot less discovery of types, and just straight forward event-driven design.